classes/output/mobile.php

```diff
@@ -0,0 +1,324 @@
+<?php
+namespace mod_groupselect\output;
+ 
+defined('MOODLE_INTERNAL') || die();
+require_once(dirname(dirname(__DIR__)).'/lib.php');
+require_once(dirname(dirname(__DIR__)).'/locallib.php');
+
+use context_module;
+use moodle_url;
+
+/**
+ * The mod_hsuforum mobile app compatibility.
+ *
+ * @package	mod_hsuforum
+ * @copyright  2018 GetSmarter {@link http://www.getsmarter.co.za}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class mobile {
+
+    public static function app_init(array $args) : array {
+        global $CFG;
+        return [
+                'templates' => [],
+                'javascript' => file_get_contents($CFG->dirroot . '/mod/groupselect/appjs/app_init.js'),
+                'otherdata' => '',
+                'files' => []
+        ];
+    }
+
+	/**
+	 * Returns the hsuforum discussion view for a given forum.
+	 * Note use as much logic and functions from view.php as possible (view.php uses renderer.php and lib.php to build view)
+	 * @param  array $args Arguments from tool_mobile_get_content WS
+	 * @return array HTML, javascript and otherdata
+	*/
+    public static function groupselect_view($args) {
+        global $OUTPUT, $USER, $DB, $PAGE;
+
+        $args = (object) $args;
+
+        $cm = get_coursemodule_from_id ( 'groupselect', $args->cmid, 0, false, MUST_EXIST );
+        $groupselect = $DB->get_record ( 'groupselect', array (
+            'id' => $cm->instance
+        ), '*', MUST_EXIST );
+
+        $course = $DB->get_record ( 'course', array ('id' => $cm->course), '*', MUST_EXIST );
+
+        require_login ( $course, true, $cm );
+        $context = context_module::instance ( $cm->id );
+
+
+        $mygroups = groups_get_all_groups ( $course->id, $USER->id, $groupselect->targetgrouping, 'g.*' );
+
+        $isopen = groupselect_is_open ( $groupselect );
+        $groupmode = groups_get_activity_groupmode ( $cm, $course );
+        $counts = groupselect_group_member_counts ( $cm, $groupselect->targetgrouping );
+        $groups = groups_get_all_groups ( $course->id, 0, $groupselect->targetgrouping );
+        $passwordgroups = groupselect_get_password_protected_groups ( $groupselect );
+        $hidefullgroups = $groupselect->hidefullgroups;
+        $exporturl = '';
+        $ASSIGNROLE = 4;
+
+        // Permissions
+        $accessall = has_capability ( 'moodle/site:accessallgroups', $context );
```

Comment:
Not sure if we should have so many open spaces and for the few lines below the same